### PR TITLE
Fat 189 mod-password-validator: Setup API Karate tests in the module

### PR DESCRIPTION
**Pull request by Igor-Gorchakov**

### CODE REVIEW

File: mod-password-validator/src/test/java/org.folio/ModPasswordValidatorApiTest.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public class ModPasswordValidatorApiTest extends TestBase {
+    private static final String TEST_BASE_PATH = "classpath:domain/passwordvalidator/features/";
+
+    public ModPasswordValidatorApiTest() {
+        super(new TestIntegrationService(new TestModuleConfiguration(TEST_BASE_PATH)));
+    }
+
+    @BeforeAll
+    public void modDataExportTestsBeforeAll() {
```
Comment (oleksandr-dekin) [inline mod-password-validator/src/test/java/org.folio/ModPasswordValidatorApiTest.java:23]: please rename method

File: mod-password-validator/src/test/java/org.folio/ModPasswordValidatorApiTest.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        runFeature("classpath:domain/passwordvalidator/basic-junit.feature");
+    }
+
+    @Test
+    void quickExportTest() {
+        runFeatureTest("rules");
+    }
+
+    @AfterAll
+    public void ordersApiTestAfterAll() {
```
Comment (oleksandr-dekin) [inline mod-password-validator/src/test/java/org.folio/ModPasswordValidatorApiTest.java:33]: please rename method

File: mod-password-validator/src/test/java/org.folio/ModPasswordValidatorApiTest.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        super(new TestIntegrationService(new TestModuleConfiguration(TEST_BASE_PATH)));
+    }
+
+    @BeforeAll
+    public void modDataExportTestsBeforeAll() {
+        runFeature("classpath:domain/passwordvalidator/basic-junit.feature");
+    }
+
+    @Test
+    void quickExportTest() {
```
Comment (oleksandr-dekin) [inline mod-password-validator/src/test/java/org.folio/ModPasswordValidatorApiTest.java:28]: please rename method

File: mod-password-validator/src/test/resources/domain/basic.feature [hunk 4]
```diff
@@ -0,0 +1,29 @@
+Feature: mod-password-validator integration tests
```
Comment (oleksandr-dekin) [inline mod-password-validator/src/test/resources/domain/basic.feature:1]: You duplicate permissions and calling destroy data(you call in `ModPasswordValidatorApiTest`). Please, remove this file and just add run feature (domain/basic.feature) in the `ModPasswordValidatorApiTest`. For example you can see from another modules: ModNotesTest and QuickMarcApiTest

File: mod-password-validator/src/test/resources/domain/passwordvalidator/basic.feature [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    * table userPermissions
+      | name         |
+      | 'okapi.proxy.tenants.post'        |
+      | 'validation.all'                  |
+      | 'login.password.validate'         |
+      | 'users.collection.get'            |
+
+  Scenario: Create tenant and users for testing
+    Given call read('classpath:common/setup-users.feature')
+    Given call login testAdmin
```
Comment (oleksandr-dekin) [inline mod-password-validator/src/test/resources/domain/passwordvalidator/basic.feature:30]: It's login need for local testing, please remove it.

File: mod-password-validator/src/test/resources/global/add-okapi-permissions.feature [hunk 6]
```diff
@@ -0,0 +1,81 @@
+Feature: prepare data for api test
```
Comment (oleksandr-dekin) [inline mod-password-validator/src/test/resources/global/add-okapi-permissions.feature:1]: This file need only for local testing, you can remove it. Generally, we run tests like that:
```
-Dtest=QuickMarcApiTest -DargLine="-Dkarate.env=testing"
```

File: mod-password-validator/src/test/resources/karate-config.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
+function fn() {
+
+  karate.configure('logPrettyRequest', true);
+  karate.configure('logPrettyResponse', true);
+
+  var env = 'localhost';
```
Comment (oleksandr-dekin) [inline mod-password-validator/src/test/resources/karate-config.js:6]: Please change to global env:
```
var env = karate.env;
```

File: mod-password-validator/src/test/resources/karate-config.js [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+    config.admin = {tenant: 'supertenant', name: 'admin', password: 'admin'}
+  }
+
+   var params = JSON.parse(JSON.stringify(config.admin))
+   params.baseUrl = config.baseUrl;
+   var response = karate.callSingle('classpath:common/login.feature', params)
+   config.adminToken = response.responseHeaders['x-okapi-token'][0]
+
+//   uncomment to run on local
+//   karate.callSingle('classpath:global/add-okapi-permissions.feature', config);
```
Comment (oleksandr-dekin) [inline mod-password-validator/src/test/resources/karate-config.js:42]: It's only for local testing, please remove

File: mod-password-validator/src/test/resources/domain/passwordvalidator/features/rules.feature [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+    * url baseUrl
+
+    * callonce login testAdmin
+    * def okapiAdminToken = okapitoken
+
+    * callonce login testUser
+    * def okapiUserToken = okapitoken
+
+    * def headersUser = { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapiUserToken)', 'Accept': 'application/json'  }
+    * def headersAdmin = { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapiAdminToken)', 'Accept': 'application/json'  }
```
Comment (oleksandr-dekin) [inline mod-password-validator/src/test/resources/domain/passwordvalidator/features/rules.feature:13]: You don't use `headersAdmin`. Please remove it

**Review by oleksandr-dekin — APPROVED** [review]

**Review by psmagin — APPROVED** [review]

**Pull request merged**